File: GeoQuiz/app/src/main/java/com/lucasleelz/geoquiz/QuizActivity.java
package com.lucasleelz.geoquiz;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

public class QuizActivity extends AppCompatActivity {

    private Button mTrueButton;

    private Button mFalseButton;

    private Button mNextButton;
    private TextView mQuestionTextView;

    private Question[] mQuestionBank = new Question[]{
            new Question(R.string.question_australia, true),
            new Question(R.string.question_oceans, true),
            new Question(R.string.question_mideast, true),
            new Question(R.string.question_africa, true),
            new Question(R.string.question_americas, true),
            new Question(R.string.question_asia, true),
    };

    private int mCurrentIndex = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_quiz);

        mTrueButton = (Button) findViewById(R.id.true_button);
        mFalseButton = (Button) findViewById(R.id.false_button);
        mNextButton = (Button) findViewById(R.id.next_button);

        mQuestionTextView = (TextView) findViewById(R.id.question_text_view);
        mQuestionTextView.setOnClickListener(view -> checkNext());
        updateQuestion();

        mTrueButton.setOnClickListener(view -> checkAnswer(true));
        mFalseButton.setOnClickListener(view -> checkAnswer(false));
        mNextButton.setOnClickListener(view -> checkNext());
    }

    private void updateQuestion() {
        int question = mQuestionBank[mCurrentIndex].getTextResId();
        mQuestionTextView.setText(question);
    }

    private void checkAnswer(boolean userPressedTrue) {
        boolean answerIsTrue = mQuestionBank[mCurrentIndex].isAnswerTrue();

        int messageResId = userPressedTrue == answerIsTrue
                ? R.string.correct_toast
                : R.string.incorrect_toast;

        Toast.makeText(this, messageResId, Toast.LENGTH_LONG).show();
    }

    private void checkNext() {
        mCurrentIndex = (mCurrentIndex + 1) % mQuestionBank.length;
        updateQuestion();
    }
}

File: GeoQuizForKotlin/app/src/main/java/com/lucaslz/geoquizforkotlin/QuizActivity.kt
package com.lucaslz.geoquizforkotlin

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.ImageButton
import android.widget.TextView
import android.widget.Toast

class QuizActivity : AppCompatActivity() {

    val questionBank = listOf<Question>(
            Question(R.string.question_australia, true),
            Question(R.string.question_oceans, false),
            Question(R.string.question_mideast, true),
            Question(R.string.question_africa, false),
            Question(R.string.question_americas, false),
            Question(R.string.question_asia, true)
    )

    var currentQuestionIndex = 0

    private var questionTextView: TextView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_quiz)

        questionTextView = findViewById<TextView>(R.id.question_text_view)
        val trueButton = findViewById<Button>(R.id.true_button)
        val falseButton = findViewById<Button>(R.id.false_button)
        val previousImageButton = findViewById<ImageButton>(R.id.previous_image_button)
        val cheatButton = findViewById<Button>(R.id.cheat_button)
        val nextImageButton = findViewById<ImageButton>(R.id.next_image_button)

        updateQuestionTextView()
        trueButton.setOnClickListener {
            checkAnswer(true)
        }

        falseButton.setOnClickListener {
            checkAnswer(false)
        }
    }

    private fun updateQuestionTextView() {
        val question = questionBank[currentQuestionIndex]
        questionTextView!!.setText(question.textResId)
    }

    private fun checkAnswer(userPressedTrue: Boolean) {
        val currentQuestion = questionBank[currentQuestionIndex]
        var messageResId = R.string.incorrect_toast
        if (userPressedTrue == currentQuestion.answerTrue) {
            messageResId = R.string.correct_toast
        }
        Toast.makeText(this@QuizActivity, messageResId, Toast.LENGTH_SHORT).show()
    }
}
